### DISCUSSION TITLE
iOS LaunchScreen light and dark mode

### DISCUSSION THREAD

**Discussion by AlleSchonWeg** [dotnet/maui#26133] [Q&A] [answered]
Hi,
i'm trying to port my Xamarin iOS Launchscreen to Maui. But it's not working. In Xamarin i had a Assets.xcassets\LaunchIcon.imageset folder with 3 files.
-Contents.json
-Logo_dark.pdf
-Logo_light.pdf
All 3 files are referenced in the csproj file like this:
```XML
   <ImageAsset Include="Assets.xcassets\LaunchIcon.imageset\Contents.json">
      <Visible>false</Visible>
    </ImageAsset>
```
and my LaunchScreen.storyboard which references the LaunchIcon imageset. I also set UILaunchStoryboardName in Info.plist.
The imageset was created in Xcode and set the Logo_dark.pdf for All Scales, All Screen, Dark. The Logo_light.pdf do the same for light.
I tried the same with maui (ImageAsset build action) but without success. I see just a white screen with no logo.
How can i create a SpashScreen for dark and light mode in Maui?
Thank you

**Comment by AlleSchonWeg** [comment, marked as answer]
After two days of try and error it finally works. The trick was to set the link property in the csproj file. Here is what i have done. 
```XML
	<ItemGroup Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'ios'">
		<ImageAsset Include="Platforms\iOS\Splash\LaunchIcon.imageset\Contents.json" Link="Assets.xcassets\LaunchIcon.imageset\Contents.json" />
		<ImageAsset Include="Platforms\iOS\Splash\LaunchIcon.imageset\Logo_dark.pdf" Link="Assets.xcassets\LaunchIcon.imageset\Logo_dark.pdf" />
		<ImageAsset Include="Platforms\iOS\Splash\LaunchIcon.imageset\Logo_light.pdf" Link="Assets.xcassets\LaunchIcon.imageset\Logo_light.pdf" />
		<InterfaceDefinition Include="Platforms\iOS\Splash\LaunchScreen.storyboard" Link="LaunchScreen.storyboard" />
	</ItemGroup>
```
Also set in info.plist
```XML
		<key>UILaunchStoryboardName</key>
		<string>LaunchScreen.storyboard</string>
```

**Reply by marevalointermedia** [reply]
hello AlleschonWeg! thanks for sharing your solution. I can't access to XCode now, and my storyboard isn't working. Can you share your storyboard xml too?

**Comment by paulober** [comment]
@AlleSchonWeg Does this still work for you?
I tried it also with:
```xml
<ImageAsset Include="Platforms\iOS\Splash\LaunchLogo.imageset\Contents.json"
                    Link="Assets.xcassets\LaunchLogo.imageset\Contents.json" />
<ImageAsset Include="Platforms\iOS\Splash\LaunchLogo.imageset\splash.pdf"
                    Link="Assets.xcassets\LaunchLogo.imageset\splash.pdf" />
<InterfaceDefinition Include="Platforms\iOS\Splash\LaunchScreen.storyboard" />
```
and in the story bod had an image view with `Image="LaunchLogo"` 
and the contents.json in the imageset looks like this:
```json
{
  "images" : [
    {
      "filename" : "splash.pdf",
      "idiom" : "universal"
    }
  ],
  "info" : {
    "author" : "xcode",
    "version" : 1
  },
  "properties" : {
    "preserves-vector-representation" : true
  }
}
```
but for some reason it only shows the background of the storyboard but not the image.

**Reply by paulober** [reply]
Found a solution. Providing the vector pdf a second time but for dark mode universal fixed it:
```json
"appearances" : [
        {
          "appearance" : "luminosity",
          "value" : "dark"
        }
      ],
```
